packages/react/src/primitives/ScrollView/ScrollView.tsx

```diff
@@ -11,23 +11,42 @@ import {
 } from '../types';
 import { View } from '../View';
 import { primitiveWithForwardRef } from '../utils/primitiveWithForwardRef';
+import { useComposeRefsCallback } from '../../hooks/useComposeRefsCallback';
 
 const ScrollViewPrimitive: Primitive<ScrollViewProps, 'div'> = (
-  { children, className, orientation, ...rest },
-  ref
-) => (
-  <View
-    className={classNames(
-      ComponentClassName.ScrollView,
-      classNameModifier(ComponentClassName.ScrollView, orientation),
-      className
-    )}
-    ref={ref}
-    {...rest}
-  >
-    {children}
-  </View>
-);
+  { children, className, orientation, autoScroll, ...rest },
+  externalRef
+) => {
+  const internalRef = React.useRef<HTMLDivElement | null>(null);
+  const composedRefs = useComposeRefsCallback<HTMLDivElement | null>({
+    externalRef,
+    internalRef,
+  });
+
+  React.useLayoutEffect(() => {
+    if (autoScroll) {
+      internalRef.current?.scrollTo({
+        top: internalRef.current?.scrollHeight,
+        left: internalRef.current?.scrollWidth,
+        behavior: autoScroll,
+      });
+    }
+  }, [children, autoScroll]);
```

Comment:
Without context in to why `children` is included in the dependency array, a future maintainer might think this was a typo and remove. 

Can we add a short comment here explaining the impact on behavior and a unit test that makes an  assertion that will fail if `children` is removed?